src/controllers/user.controller.ts

```diff
@@ -430,3 +434,89 @@ export const declineFriendRequest = async (req: Request, res: Response) => {
     return res.status(500).send({ message: error?.message });
   }
 };
+
+/**
+ * Uploads a profile image for the user with the specified userId to cloudinary
+ * and updates the user's field to point to the new image.
+ *
+ * DIMENSIONS:    automatically resizes the image to the specified dimensions
+ * USER_FIELDS:   the MongoDB fields in the user document that will be updated with the new image url
+ *                of the resized image
+ *
+ * NOTE:          the order of the dimensions and user fields must match
+ *
+ * @param req - the Request object containing the userId in the params and the image in the body
+ * @param res - the Response object sent back to the client
+ * @returns Returns either an error response with a 400 or 500 status code and a message,
+ * or a success response with a 200 status code and a message
+ */
+export const uploadProfileImg = async (req: Request, res: Response) => {
+  const DIMENSIONS = [128, 256];
+  const USER_FIELDS = ['imgUrlProfileSmall', 'imgUrlProfileLarge'];
```

Comment:
also consider a single array of objects (with dimensions field and user field?) so that order doesn't matter